xcodeproj/internal/xcode_schemes.bzl

```diff
@@ -276,11 +276,49 @@ def make_xcode_schemes(bazel_labels):
             build_for = build_for,
         )
 
+    def _sanitizers(
+            address = False,
+            thread = False,
+            undefined_behavior = False):
+        """Constructs the scheme's sanitizers' default state. The state can also be modified in Xcode.
+
+        Args:
+            address: Optional. A boolean value representing
+                whether the address sanitizer should be enabled or not.
+            thread: Optional. A boolean value representing
+                whether the thread sanitizer should be enabled or not.
+            undefined_behavior: Optional. A boolean value representing
+                whether the undefined behavior sanitizer should be enabled or not.
+        """
+        if address and thread:
+            fail("Address Sanitizer cannot be used together with Thread Sanitizer.")
+        return struct(
+            address = address,
+            thread = thread,
+            undefined_behavior = undefined_behavior,
+        )
+
+    def _diagnostics(
+            sanitizers = None):
+        """Constructs the scheme's diagnostics.
+
+        Args:
+            sanitizers: Optional. A `struct` value as created by
+                `xcode_schemes.sanitizers`.
+
+        Returns:
+            A `struct` representing scheme's diagnostics.
+        """
+        return struct(
+            sanitizers = sanitizers,
+        )
+
     def _launch_action(
             target,
             args = None,
             env = None,
-            working_directory = None):
+            working_directory = None,
+            diagnostics = None):
```

Comment:
How does the `Decodable` handle the `None` here?

Reply:
We have kept `diagnostics` as an optional property in the generator so if diagnostics is `None`, diagnostics will be `nil` when decoded and the sanitizers default state(in the generated scheme) will be `False`.